src/main/java/com/google/devtools/build/lib/bazel/coverage/CoverageReportActionBuilder.java

```diff
@@ -242,18 +243,20 @@ private CoverageReportAction generateCoverageReportAction(
     Artifact lcovOutput = args.factory().getDerivedArtifact(
         coverageDir.getRelative("_coverage_report.dat"), root, args.artifactOwner());
     Artifact reportGeneratorExec = args.reportGenerator().getExecutable();
+    RunfilesSupport runfilesSupport = args.reportGenerator().getRunfilesSupport();
     args = CoverageArgs.createCopyWithCoverageDirAndLcovOutput(args, coverageDir, lcovOutput);
     ImmutableList<String> actionArgs = argsFunc.apply(args);
 
-    ImmutableList<Artifact> inputs = ImmutableList.<Artifact>builder()
+    ImmutableList.Builder<Artifact> inputsBuilder = ImmutableList.<Artifact>builder()
         .addAll(args.coverageArtifacts())
         .add(reportGeneratorExec)
-        .add(args.reportGenerator().getRunfilesSupport().getRunfilesMiddleman())
-        .add(args.lcovArtifact())
-        .build();
+        .add(args.lcovArtifact());
+    if (runfilesSupport != null) {
```

Comment:
We need to check if the following are non-null:
- `args.reportGenerator()`
- `args.reportGenerator().getRunfilesSupport()`
- `args.reportGenerator().getRunfilesSupport().getRunfilesMiddleman()`

so better replace with:

```
if (args.reportGenerator() != null && args.reportGenerator().getRunfilesSupport() != null && args.reportGenerator().getRunfilesSupport().getRunfilesMiddleman() != null) {
    inputsBuilder.add(args.reportGenerator().getRunfilesSupport().getRunfilesMiddleman());
}
```

Reply:
Ack. I'm really sorry that you're dealing with summer intern-level mistakes. :(

I assumed that `reportGenerator` and `getRunsfilesMiddleman` were safe by virtue of not having the `@Nullable` annotation (cf. `CoverageArgs.coverageDir` and `CoverageArgs.lcovOutput`).

Also, `reportGeneratorExec` is assigned to `args.reportGenerator().getExecutable()` w/o checking `reportGenerator() != null`. Does that need a fixup as well?

Reply:
You were right actually, the check you wrote was enough. Sorry about that! I'm currently importing the change.